Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/nbclient/client.py", line 1305, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/jupyter_core/utils/__init__.py", line 165, in wrapped
    return loop.run_until_complete(inner)
  File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()
  File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/nbclient/client.py", line 705, in async_execute
    await self.async_execute_cell(
  File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/nbclient/client.py", line 1058, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/nbclient/client.py", line 914, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
#Vary umbrella potential from 0 to 1000. Study how Tc affects the U_k and U_m 

S, E, M = run_ising2d_umbrella(spins=np.random.choice([-1, 1], size=(20, 20)), 
                T=1, 
                J=1, 
                B=0, 
                U_k = 0,
                U_m = 0.5, 
                n_steps=2000000, 
                out_freq=10)

plot_US(S,E, M)
------------------


[0;31m---------------------------------------------------------------------------[0m
[0;31mNameError[0m                                 Traceback (most recent call last)
Cell [0;32mIn[3], line 12[0m
[1;32m      1[0m [38;5;66;03m#Vary umbrella potential from 0 to 1000. Study how Tc affects the U_k and U_m [39;00m
[1;32m      3[0m S, E, M [38;5;241m=[39m run_ising2d_umbrella(spins[38;5;241m=[39mnp[38;5;241m.[39mrandom[38;5;241m.[39mchoice([[38;5;241m-[39m[38;5;241m1[39m, [38;5;241m1[39m], size[38;5;241m=[39m([38;5;241m20[39m, [38;5;241m20[39m)), 
[1;32m      4[0m                 T[38;5;241m=[39m[38;5;241m1[39m, 
[1;32m      5[0m                 J[38;5;241m=[39m[38;5;241m1[39m, 
[0;32m   (...)[0m
[1;32m      9[0m                 n_steps[38;5;241m=[39m[38;5;241m2000000[39m, 
[1;32m     10[0m                 out_freq[38;5;241m=[39m[38;5;241m10[39m)
[0;32m---> 12[0m [43mplot_US[49m[43m([49m[43mS[49m[43m,[49m[43mE[49m[43m,[49m[43m [49m[43mM[49m[43m)[49m

Cell [0;32mIn[2], line 2[0m, in [0;36mplot_US[0;34m(S, E, M)[0m
[1;32m      1[0m [38;5;28;01mdef[39;00m[38;5;250m [39m[38;5;21mplot_US[39m(S, E, M):
[0;32m----> 2[0m     fig, ax  [38;5;241m=[39m [43mplt[49m[38;5;241m.[39msubplots(ncols[38;5;241m=[39m[38;5;241m4[39m, figsize[38;5;241m=[39m([38;5;241m16[39m,[38;5;241m4[39m))
[1;32m      4[0m     ax[[38;5;241m0[39m][38;5;241m.[39mset(xlabel[38;5;241m=[39m[38;5;124m'[39m[38;5;124mN steps[39m[38;5;124m'[39m, ylabel[38;5;241m=[39m[38;5;124mr[39m[38;5;124m'[39m[38;5;124m$E$[39m[38;5;124m'[39m)
[1;32m      6[0m     ax[[38;5;241m0[39m][38;5;241m.[39mplot(E)

[0;31mNameError[0m: name 'plt' is not defined

